### PR TITLE
Tests for overfit on skewed classification datasets

### PR DESCRIPTION
**Pull request by rbharath**
Will add details once tests pass and cleaned up.

### CODE REVIEW

**Comment by rbharath** [comment]
Changes in this PR:

- Some fixes to `Metric` to handle to mean/median versions of standard metrics correctly.
- Add tests to `test_overfit` for skewed classification datasets
- Simplify `test_overfit` by removing unnecessary temp files.
- Add examples to apply basic models on MUV dataset (very skewed dataset). The sklearn model does fine, but the keras/tensorflow models have bad train-set performance, despite the loss dropping to zero. There is some bug here... I hoped the skewed tests in `test_overfit` would catch the bug, but no luck and the issue looks to be trickier.
